coreclient/src/user_handles/connection_packages.rs

```diff
@@ -0,0 +1,122 @@
+// SPDX-FileCopyrightText: 2025 Phoenix R&D GmbH <hello@phnx.im>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use phnxcommon::{
+    crypto::ConnectionDecryptionKey,
+    identifiers::UserHandle,
+    messages::connection_package::{ConnectionPackage, ConnectionPackageHash},
+};
+use sqlx::{Result, SqliteConnection, query, query_scalar};
+
+pub trait StorableConnectionPackage: Sized {
```

Comment:
1. Can be only `pub(crate)`.
2. Just out of curiosity, this trait is only implemented once. Is there a plan to implement it on other types? For other types persistence, we usually implements methods directly on the type, right?

Reply:
I changed the trait to require `Borrow<ConnectionPackage>`, which allows us to avoid the additional implementation. I don't think it gets more concise than this.